matchbox_socket/src/webrtc_socket/socket.rs

```diff
@@ -166,70 +183,60 @@ impl WebRtcSocket {
                 id_rx,
                 messages_from_peers,
                 peer_messages_out: peer_messages_out_tx,
-                new_connected_peers,
-                disconnected_peers,
-                peers: vec![],
+                peer_state_rx,
+                peers: Default::default(),
             },
             Box::pin(run_socket(
                 id_tx,
                 config,
                 peer_messages_out_rx,
-                new_connected_peers_tx,
-                disconnected_peers_tx,
+                peer_state_tx,
                 messages_from_peers_tx,
             )),
         )
     }
 
-    /// Returns a future that resolves when the given number of peers have connected
-    pub async fn wait_for_peers(&mut self, peers: usize) -> Vec<PeerId> {
-        debug!("waiting for peers to join");
-        let mut addrs = vec![];
-        while let Some(id) = self.new_connected_peers.next().await {
-            addrs.push(id.clone());
-            if addrs.len() == peers {
-                debug!("all peers joined");
-                self.peers.extend(addrs.clone());
-                return addrs;
-            }
-        }
-        panic!("Signal server died")
-    }
-
     /// Check if new peers have connected and if so add them as peers
-    pub fn accept_new_connections(&mut self) -> Vec<PeerId> {
-        let mut ids = Vec::new();
-        while let Ok(Some(id)) = self.new_connected_peers.try_next() {
-            self.peers.push(id.clone());
-            ids.push(id);
+    // todo: think about name?
+    pub fn handle_peer_changes(&mut self) -> Vec<(PeerId, PeerState)> {
```

Comment:
`peer_state_changes` > `handle_peer_changes` imo.

Reply:
yeah, kind of struggling with this one, I wanted the method name to make it clear that mutates it stuff (`self.connected_peers`).

Other names I considered:

- `process_peer_state_changes`
- `process_peer_changes`
- `update_peers`

Reply:
Update peers